subprojects/resources/src/main/java/org/gradle/internal/resource/local/ModificationTimeFileAccessTimeJournal.java

```diff
@@ -17,11 +17,17 @@
 package org.gradle.internal.resource.local;
 
 import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.attribute.FileTime;
 
 public class ModificationTimeFileAccessTimeJournal implements FileAccessTimeJournal {
     @Override
     public void setLastAccessTime(File file, long millis) {
-        file.setLastModified(millis);
+        try {
+            Files.setLastModifiedTime(file.toPath(), FileTime.fromMillis(millis));
+        } catch (IOException ignore) {
```

Comment:
Maybe log at debug level so we know it's happening when analysing a problem?

Reply:
Good point! Done in 9c22aac.